src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -11710,7 +11710,7 @@ int Assembler::vex_prefix_and_encode(int dst_enc, int nds_enc, int src_enc, VexS
     }
   }
 
-  if (UseAVX > 2) {
+  if (UseAVX > 2 && !attributes->uses_vl()) {
```

Comment:
This is already coved by below assertion

Reply:
Without this check, the test is failing for KNL (-XX:UseAVX=3 -XX:+UnlockDiagnosticVMOptions -XX:+UseKNLSetting) as Vladimir mentioned. Is there a better way to handle the KNL case?

Reply:
Yes, there is an easy way. For the instructs where you added the pxor instruction generation, you could change the dst register type from regF to  vlRegF. This restricts the xmm register to xmm0-xmm15 for KNL, thereby not needing the evex encoding and in-turn not needing the avx512vl support for pxor.

Reply:
and likewise from regD to vlRegD.

Reply:
> Yes, there is an easy way. For the instructs where you added the pxor instruction generation, you could change the dst register type from regF to vlRegF. 

Thanks Sandhya, will make the changes and push an update.